### PR TITLE
Solcast service

### PR DESCRIPTION
**Pull request by Malte96**
- solcast-service auf Basis des ownServices erstellt
- Möglichkeit Vorgabe Zeitstempel aus Datenquelle für _send_data_message eingepflegt

### CODE REVIEW

File: tek_box_software/services/basic_service.py [hunk 1]
```diff
@@ -329,12 +330,13 @@ async def handle_command_message(self, cmd: Dict, options: Optional[Dict]) -> RE
...
         return RETURN_CODE.SUCCESS
 
-    async def create_measurment_tasks(self):
+    async def create_measurement_tasks(self):
         """
         init the devices and create the tasks for the measurment
         """
         interface = self._get_service_config()
         accesspoints = interface.get("accesspoints", [])
+        logger.info("create_measurement_tasks - basic")
```
Comment (karlTGA) [inline tek_box_software/services/basic_service.py:339]: please remove this comment

File: tek_box_software/services/solcast/service.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        time_datapoint = accesspoint.get("time_datapoint")
+        request_type = accesspoint.get("request_type")
+        accesspoint_label = accesspoint["label"]
+        interface = self._get_service_config()
+        group_label = interface.get("type")
+
+        try:
+            with urlopen(solcast_api_url) as response:
+                weather_data = json.load(response)
+            for time in weather_data[request_type]:
```
Comment (karlTGA) [inline tek_box_software/services/solcast/service.py:86]: is time really a good name for the entry in the data?

File: tek_box_software/services/solcast/service.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+class SolcastActualsService(SolcastService):
+    SERVICE_TYPE = "solcast-actuals"
+
+    def __init__(
+        self,
+        default_measurement_interval: int = 10,
+    ) -> None:
+        super().__init__(default_measurement_interval=default_measurement_interval)
+
+        self._host = None
```
Comment (karlTGA) [inline tek_box_software/services/solcast/service.py:114]: What is that?

**Review by karlTGA — CHANGES_REQUESTED** [review]
Some small issues. But all in all fine.

File: tek_box_software/services/basic_service.py (hunk 1 above)
Comment (Malte96) [reply tek_box_software/services/basic_service.py:339]: I will take out, I have only missed this.

File: tek_box_software/services/solcast/service.py (hunk 2 above)
Comment (Malte96) [reply tek_box_software/services/solcast/service.py:86]: I will find a better name.

File: tek_box_software/services/solcast/service.py (hunk 3 above)
Comment (Malte96) [reply tek_box_software/services/solcast/service.py:114]: Not really necessary, was based on owm service.

File: tek_box_software/services/solcast/service.py [hunk 4]
```diff
@@ -82,12 +82,12 @@ async def _measure(self, accesspoint: AccesspointSolcastConfig, solcast_api_url:
 
         try:
             with urlopen(solcast_api_url) as response:
-                weather_data = json.load(response)
-            for time in weather_data[request_type]:
-                timestamp = datetime.strptime(time[time_datapoint], "%Y-%m-%dT%H:%M:%S.%f0Z")
+                solcast_response_data = json.load(response)
+            for data_timestap in solcast_response_data[request_type]:
```
Comment (karlTGA) [inline tek_box_software/services/solcast/service.py:86]: wat is a timestap? ^^

**Review by karlTGA — APPROVED** [review]
only type, rest is fine

File: tek_box_software/services/solcast/service.py (hunk 4 above)
Comment (Malte96) [reply tek_box_software/services/solcast/service.py]: oh, awkward typo

**Pull request merged**